### ISSUE TITLE
OpenCV 4.7.0 much slower than 4.6.0 for Java Apps

### ISSUE THREAD

**Issue by tom131313** [opencv/opencv#23590]
### System Information

Performance comparisons are between OpenCV 4.7.0 (slow) and 4.6.0. (faster). [4.5.0 runs essentially same faster speed as 4.6.0.]
All runs use Windows 10; HP EliteBook laptop.

My test app (the MRE) built with Java compiler:
openjdk version "17.0.5" 2022-10-18
OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)
OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (build 17.0.5+8, mixed mode, sharing)

Essentially identical results running test on two Java runtime versions:

java version "12" 2019-03-19
Java(TM) SE Runtime Environment (build 12+33)
Java HotSpot(TM) 64-Bit Server VM (build 12+33, mixed mode, sharing)
  and
openjdk version "17.0.5" 2022-10-18
OpenJDK Runtime Environment Temurin-17.0.5+8 (build 17.0.5+8)
OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (build 17.0.5+8, mixed mode, sharing)

### Detailed description

OpenCV 4.7.0 is much slower than 4.6.0 or 4.5.0 for the test run as shown using `imread` in the test MRE, for a variety of other methods in another app summarized below, and for other Java apps using OpenCV.
```
MRE Run times OpenCV 4.6.0 [seconds]
  2000 `imread` elapsed time 3.339, total CPU time 2.984375, user CPU 2.0625

MRE Run times OpenCV 4.7.0 [seconds]
  2000 `imread` elapsed time 6.915, total CPU time 6.40625, user CPU 5.640625
```
The MRE test case scales correctly as expected:
2000 iterations takes twice as long as 1000 iterations.

MRE Run times OpenCV 4.6.0 with `imread` commented out are nearly zero [seconds]:
  2000 `imread` elapsed time 0.014, total CPU time 0.0, user CPU 0.015625

Instrumenting a complete App showed similar performance differences for most OpenCV Java methods (4.5.0 and 4.6.0 as are essentially identical in performance).

```
              [version/milliseconds]
                 4.5.0       4.7.0
                 _____       _____
 imread          69943      173004
 convert          9054      324540
 resize           2090       44447
 blur                0           0 
 split              78         120
 normalize          21         648
 equalize           25          77
 resize             14          22
 adaptive threshold 14          64
```

### Steps to reproduce

A MRE app with the single method, `imread` was written to demonstrate the slow performance. It is attached and includes generating its own test file to read and it displays the OpenCV version build information for both versions which is included as comments in the file.
[App.java.txt](https://github.com/opencv/opencv/files/11423955/App.java.txt)

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [X] There is reproducer code and related data files (videos, images, onnx, etc)

**Comment by dkurt** [comment]
Can you provide a build summary for both versions by `getBuildInformation()`?

**Comment by dkurt** [comment]
@tom131313, At least compilers are different, do you have an option to compile both versions with the same one?
```
General configuration for OpenCV 4.6.0 =====================================
  Version control:               4.6.0

  Platform:
    Timestamp:                   2022-06-05T15:49:50Z
    Host:                        Windows 10.0.16299 AMD64
    CMake:                       3.20.21032501-MSVC_2
    CMake generator:             Visual Studio 14 2015 Win64
    CMake build tool:            MSBuild.exe
    MSVC:                        1900
    Configuration:               Debug Release
```
```
General configuration for OpenCV 4.7.0-dev =====================================
  Version control:               4.7.0-1-g9208dcb07c

  Platform:
    Timestamp:                   2022-12-28T16:14:24Z
    Host:                        Windows 10.0.19045 AMD64
    CMake:                       3.23.3
    CMake generator:             Visual Studio 16 2019
    CMake build tool:            C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/MSBuild/Current/Bin/MSBuild.exe
    MSVC:                        1928
    Configuration:               Debug Release
```

**Comment by tom131313** [comment]
I don't know how to make or compile OpenCV. I don't think I have a C++ compiler and certainly not any other dependencies. About the only help I could be is I'm glad to run Java test cases, if a new OpenCV jar and dll zip/exe file is sent to me.

Also, there is a possibility that the two different Java compilers OpenCV used is the trouble but it seems unlikely since the `imread` is only one line of code that calls the native C++ code:
```
public static Mat imread(String filename, int flags) {
        return new Mat(imread_0(filename, flags));
}
```

**Comment by dkurt** [comment]
@tom131313, your problem with `imread` can be related to `libjpeg-turbo`. See the following discussions: https://github.com/opencv/opencv/issues/22472, SIMD is not enabled:

```
General configuration for OpenCV 4.7.0-dev =====================================
  Version control:               4.7.0-1-g9208dcb07c
...
    JPEG:                        build-libjpeg-turbo (ver 2.1.3-62)
      SIMD Support Request:      YES
      SIMD Support:              NO
```

**Comment by opencv-alalek** [comment]
@tom131313 Could you please try this package of 4.7.0 from OpenCV legacy CI?
- I exported package archive here: http://pullrequest.opencv.org/buildbot/export/tmp/20221228-151929-725e440d278aca07d35a5e8963ef990572b07316/  (`opencv-master-winpack.exe` is `opencv-4.7.0-vc14_vc15.exe` after rename)
- Packaging job (for logs reference): http://pullrequest.opencv.org/buildbot/builders/4_x_winpack_controller/builds/100014

MD5 check sum:
```
$ md5sum *
1278e914cbcb3fd16169ad9ba2ad9524  opencv-master-winpack.exe
```

**Comment by tom131313** [comment]
@opencv-alalek Thank you. Your new version of 4.7.0 runs the simple test App almost as fast as the 4.6.0 (a few percent slower, I'd guess) and that's about twice as fast as the previous 4.7.0. Another complicated App (example of change camera perspective with slider bars) performs quickly, too.

There is an inexplicable (to me) problem, though. My App can use the laptop built-in camera or a USB external camera. Previously they operated with this index numbering scheme:
```
        int camera = 1; //  0 is internal, 1 is external

        VideoCapture cap;
        cap = new VideoCapture();
        cap.open(camera);
```
Today, for some reason, that numbering scheme that worked for years is now reversed with all the OpenCV versions - not just the new one you gave me. I don't understand the relationship between the OpenCV index and the operating system managing the cameras but it changed and I doubt OpenCV caused it.

**Comment by asmorkalov** [comment]
I'll check Github Actions and BuildBot build options to fix the degradation for 4.8.0 release.

**Comment by asmorkalov** [comment]
Fix: https://github.com/opencv/ci-gha-workflow/pull/99
